test/quantization/test_quantize_jit.py

```diff
@@ -2928,14 +2929,14 @@ def forward(self, indices1, offsets1, indices2, offsets2):
                 m = torch.jit.trace(module, dummy_inputs)
             else:
                 m = torch.jit.script(module)
-            from torch.quantization import QConfigDynamic, PlaceholderObserver
-            int4_dynamic_qconfig = QConfigDynamic(activation=PlaceholderObserver.with_args(dtype=torch.float,
-                                                                                           custom_op_name="embedding_bag_4bit"),
-                                                  weight=PlaceholderObserver.with_args(custom_op_name="embedding_bag_4bit"))
-            int8_dynamic_qconfig = QConfigDynamic(activation=PlaceholderObserver.with_args(dtype=torch.float,
-                                                                                           custom_op_name="embedding_bag_byte"),
-                                                  weight=PlaceholderObserver.with_args(custom_op_name="embedding_bag_byte"))
-            m = quantize_dynamic_jit(m, {'embedding1' : int4_dynamic_qconfig, 'embedding2' : int8_dynamic_qconfig})
+            int4_qconfig = QConfig(activation=PlaceholderObserver.with_args(dtype=torch.float,
+                                                                            custom_op_name="embedding_bag_4bit"),
+                                   weight=PlaceholderObserver.with_args(custom_op_name="embedding_bag_4bit"))
+            int8_qconfig = QConfig(activation=PlaceholderObserver.with_args(dtype=torch.float,
+                                                                            custom_op_name="embedding_bag_byte"),
+                                   weight=PlaceholderObserver.with_args(custom_op_name="embedding_bag_byte"))
+            m = prepare_jit(m, {'embedding1' : int4_qconfig, 'embedding2' : int8_qconfig})
```

Comment:
What about eager mode? Currently we expose embedding bag as a module in torch/nn/quantized/dynamic. Should we change that too?. Strictly speaking this case straddles the boundary of static vs dynamic:  Output activations are in fp32 (like dynamic). Inputs are addresses and require no quantization.  Weights alone are quantized.

Reply:
I'm not sure about eager mode. Currently for static quant we expect the users to provide calibration fn when they call `quantize`. Weight only quantization shouldn't require that step. So in that sense it fits better into dynamic.